python-lib/prologin/matchmaker/matchmaker.py

```diff
@@ -131,7 +133,8 @@ def run(self, *args, **kwargs):
             self.config["matchmaker"]["port"],
         )
 
-        asyncio.create_task(self.run_task())
+        loop = asyncio.get_event_loop()
+        loop.create_task(self.run_task())
```

Comment:
Why is that needed? In modern asyncio you should never have to use loops manually when using the high level API.

Reply:
We cannot use `create_task()` here because there is no loop when this function runs and it raises `RuntimeError`.

In [other services](https://github.com/prologin/sadm/blob/276ed79b53d53cd0b486b302b942e0442ce57e2b/prologin/masternode/master.py#L59) we use `asyncio.Task()`, but that's discouraged too.

We cannot make `run()` a coro and use `asyncio.run(s.run())` either because that clashes with [aiohttp's model](https://github.com/aio-libs/aiohttp/blob/5f0a59fd38f048ee65b6199a26d2355075d0d196/aiohttp/web.py#L447), which expects to start the loop with `run_until_complete()`.

I'm now proposing an alternate solution, which uses aiohttp's [`on_startup` signal](https://docs.aiohttp.org/en/stable/web_advanced.html#aiohttp-web-signals) to start the service once the web app is started.

We may want to consider refactoring the other services with this pattern.

Reply:
Hmm, I don't think the other services would have the same issue, as the loop is already running when these tasks are created.

Reply:
Are you sure? In https://github.com/prologin/sadm/blob/master/prologin/masternode/master.py#L57-L66 the loop is started in `super().run()`.

Reply:
Indeed, I stand corrected!